.concourse/pipelines/develop-commit.yaml

```diff
@@ -187,7 +187,6 @@ jobs:
               run:
                 path: bash
                 args: ["git-polymesh/.concourse/scripts/t-npm-test.sh", "git-polymesh", "npm-caches", "artifact"]
-            timeout: 30m
```

Comment:
worth keeping in, but doubling/tripling the timeout instead?

Reply:
ah, I think I see your reasoning; `npm` later on doesn't respond to the exit signal?

Reply:
The timeout was moved into `.concourse/scripts/t-npm-test.sh`.  The reason for this is that the task timeout starts when the task begins, but the actual container may not start running until later.  The reason for **that** is that we have a limit on how many containers can run concurrently per worker (currently 2) to prevent resource exhaustion when multiple builds are triggered in a short time frame.  As a consequence the task may be stuck waiting for an available worker for some time, during which the task timeout clock is ticking.  Moving the timeout into the script and invoking `timeout 30m npm test` instead of just `npm test` works around that issue.

Reply:
PS: It's not that `npm` does not respond to the exit signal, it's that the test framework tries to recover from errors instead of reporting it with an exit code, causing the tests to idle.